src/app/tests/TestReadInteraction.cpp

```diff
@@ -1710,12 +1711,36 @@ void TestReadInteraction::TestSubscribeUrgentWildcardEvent(nlTestSuite * apSuite
         NL_TEST_ASSERT(apSuite, engine->GetNumActiveReadHandlers(ReadHandler::InteractionType::Subscribe) == 1);
 
         GenerateEvents(apSuite, apContext);
-        NL_TEST_ASSERT(apSuite, !delegate.mpReadHandler->mFlags.Has(ReadHandler::ReadHandlerFlags::HoldReport));
+        NL_TEST_ASSERT(apSuite, delegate.mpReadHandler->mFlags.Has(ReadHandler::ReadHandlerFlags::HoldReport));
         NL_TEST_ASSERT(apSuite, delegate.mpReadHandler->IsDirty() == true);
         delegate.mGotEventResponse = false;
         delegate.mGotReport        = false;
-        ctx.DrainAndServiceIO();
-        NL_TEST_ASSERT(apSuite, delegate.mGotEventResponse);
+
+        // wait for min interval 2 seconds(in test, we use 1.9second considering the time variation), expect no event is received,
+        // then wait for 0.5 seconds, then the urgent event would be sent out
+        while (true)
```

Comment:
Minor nit: Why couldn't this have just used `DriveIOUntil`?

Reply:
Ah, so we cannot use `DriveIOUntil` because it will drive the IO _at least once_. This is a problem that is an issue with this logic here too. If by the time you get to this while loop, more than 1.9s has elapsed, you'll still end up driving the IO which will falsely set `mGotEventResponse` to true, failing the test.

What you want is to only call DriveIO if the time has not elapsed.